### PR TITLE
docs(json-threat-protection): add docs for json-threat-protection plugin

### PR DESCRIPTION
**Pull request by Water-Melon**
### Testing instructions

Preview link: https://deploy-preview-7747--kongdocs.netlify.app/hub/kong-inc/json-threat-protection/unreleased/

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

KAG-4699

### CODE REVIEW

File: app/_hub/kong-inc/json-threat-protection/how-to/_index.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  --data "config.max_object_entry_count=2" \
+  --data "config.max_object_entry_name_length=3" \
+  --data "config.max_array_element_count=2" \
+  --data "config.max_string_value_length=3" \
+  --data "config.enforce_mode=block" \
+  --data "config.error_status_code=400" \
+  --data "config.error_message=BadRequest1"
+```
+
+{$ endnavtab %}
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/how-to/_index.md:28]:
```suggestion
{% endnavtab %}
```

File: app/_hub/kong-inc/json-threat-protection/how-to/_index.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/how-to/_index.md:47]:
```suggestion
{% endnavtab %}
```

File: app/_hub/kong-inc/json-threat-protection/how-to/_index.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/how-to/_index.md:66]:
```suggestion
{% endnavtab %}
```

File: app/_hub/kong-inc/json-threat-protection/how-to/_index.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+{ % endnavtabs %}
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/how-to/_index.md:68]:
```suggestion
{% endnavtabs %}
```

File: app/_hub/kong-inc/json-threat-protection/how-to/_index.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
+---
+nav_title: Basic config examples
+title: Basic config examples
+---
+
+## Basic configuration examples
+
+The following examples provide some typical configurations for enabling the `json-threat-protection` plugin on a [service]({{ site.links.web }}/gateway/api/admin-ee/latest/#/Services).
+
+{ % navtabs %}
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/how-to/_index.md:10]:
```suggestion
{% navtabs %}
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
+---
+nav_title: Overview
+---
+
+The JSON Threat Protection plugin provides security validation on various aspects of JSON to ensure that the incoming JSON in requests adheres to the security policy limits.
+
+It will check the following points:
```
Comment (veenaraja) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:7]: Suggest following reword.

The JSON Threat Protection plugin provides security validation against various aspects of JSON structure. The plugin validates the incoming JSON request body to ensure the payload adheres to the policy limits. Requests violating the policy will be considered malicious. The plugin can be configured to drop such requests thus protecting it from reaching the service. Optionally, the plugin can operate in tap mode to monitor the traffic

The plugin will check the following limits:

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+
+### Create a Service
+
+```bash
+curl -i -s -X POST http://localhost:8001/services \
+  --data name=example_service \
+  --data url='http://localhost/'
+```
+
+In this example, an Nginx instance is started locally with the default port 80.
```
Comment (veenaraja) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:48]: User wont know what your end goal is. It needs explanation. See my comment under the 'Using the plugin'. This statement about starting nginx needs to come before we create service and route on Kong

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+
+- Maximum container depth: 2
+- Maximum number of array elements: 2
+- Maximum number of object entries: 4
+- Maximum length of object keys: 7 (`parents`)
+- Maximum length of strings: 6 (`Joseph`)
+
+
+
+## Using the plugin
```
Comment (veenaraja) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:34]: In this example, we will first deploy a service listening on port 80. Next we will create a service in Kong and point to the service we just deployed. Then, we will create a route for this service and enforce payload limits by enabling the JSON-threat protection policy on this route. We'll configure the plugin to reject violating requests.  

Kong inspects the incoming requests, validates the payload adheres to the limits and rejects rejects requests that violate the policy.

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+  "instance_name": null,
+  "enabled": true
+}
+```
+
+
+
+## Examples:
+
+This request will successfully fetch the default page from the upstream Nginx.
```
Comment (veenaraja) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:181]: Since our intention here is to show that valid requests make it to backend, I would suggest we configure nginx to return 200 ok instead of the default page.  And then we reword this to say,
This request is successfully proxied to returns a response back

**Review by veenaraja — COMMENTED** [review]
@Water-Melon I would also suggest to highlight/mention other features of the plugin such as setting payload limit, tap mode

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md (hunk 6 above)
Comment (Water-Melon) [reply app/_hub/kong-inc/json-threat-protection/overview/_index.md]: I’ve updated the doc and added some additional content.

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md (hunk 7 above)
Comment (Water-Melon) [reply app/_hub/kong-inc/json-threat-protection/overview/_index.md]: I just simply removed this sentence.

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 10]
```diff
@@ -0,0 +1,238 @@
+---
+nav_title: Overview
+---
+
+The JSON Threat Protection plugin provides security validation against various aspects of JSON structure. The plugin validates the incoming JSON request body to ensure the payload adheres to the policy limits, regardless of whether the `Content-Type` header exists or is set to `application/json`. Requests violating the policy will be considered malicious. The plugin can be configured to drop such requests thus protecting it from reaching the service. Optionally, the plugin can operate in tap mode to monitor the traffic.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:5]:
```suggestion
The JSON Threat Protection plugin provides security validation against various aspects of JSON structure. 
The plugin validates the incoming JSON request body to ensure the payload adheres to the policy limits, regardless of whether the `Content-Type` header exists or is set to `application/json`. 
Requests violating the policy are considered malicious. 
You can configure the plugin to drop such requests, which stops them from reaching the service. 
Optionally, the plugin can operate in tap mode to monitor the traffic.
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+
+The plugin will check the following limits:
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:7]:
```suggestion

## How it works

The plugin checks the following limits:
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+}
+```
+
+- Maximum container depth: 2
+- Maximum number of array elements: 2
+- Maximum number of object entries: 4
+- Maximum length of object keys: 7 (`parents`)
+- Maximum length of strings: 6 (`Joseph`)
+
+**Note**: Length calculation for JSON strings and object entry names is based on UTF-8 characters, not bytes.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:32]:
```suggestion
{:.note}
> **Note**: Length calculation for JSON strings and object entry names is based on UTF-8 characters, not bytes.
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+
+
+Additionally, we have added a limit on the request body size, which can effectively reduce the resource overhead and potential attack risks associated with excessively large bodies.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:35]:
```suggestion
Additionally, there is a limit on the request body size, which can effectively reduce the resource overhead and potential attack risks associated with excessively large bodies.
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+
+## Using the plugin
+
+In this example, we will first deploy a service listening on port 80. When accessed successfully, it will respond with `200 OK`. Next we will create a service in Kong and point to the service we just deployed. Then, we will create a route for this service and enforce payload limits by enabling the JSON-threat protection policy on this route. We'll configure the plugin to reject violating requests.
+
+Kong inspects the incoming requests, validates the payload adheres to the limits and rejects requests that violate the policy.
+
+
+
+### Create a Service
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:46]:
```suggestion
### Create a service

Create a {{site.base_gateway}} service pointing to the external service that you want to protect. 
In this example, we're just using a local `example_service`:
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+  "tags": null,
+  "client_certificate": null,
+  "retries": 5,
+  "path": "/"
+}
+```
+
+
+
+### Create a Route
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:81]:
```suggestion
### Create a route

Create a route within {{site.base_gateway}} and attach it to the `example_service`.
Having a route allows you to set protection rules.
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 16]
```diff
@@ -0,0 +1,238 @@
...
+  "tags": null,
+  "strip_path": true,
+  "request_buffering": true,
+  "response_buffering": true
+}
+```
+
+
+
+### Create a JSON-Threat-Protection Policy
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:126]:
```suggestion
### Create a JSON threat protection policy

Enable the `json-threat-protection` plugin on the route you just created:
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 17]
```diff
@@ -0,0 +1,238 @@
...
+
+**Note**: Length calculation for JSON strings and object entry names is based on UTF-8 characters, not bytes.
+
+
+Additionally, we have added a limit on the request body size, which can effectively reduce the resource overhead and potential attack risks associated with excessively large bodies.
+
+
+## Using the plugin
+
+In this example, we will first deploy a service listening on port 80. When accessed successfully, it will respond with `200 OK`. Next we will create a service in Kong and point to the service we just deployed. Then, we will create a route for this service and enforce payload limits by enabling the JSON-threat protection policy on this route. We'll configure the plugin to reject violating requests.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:40]:
```suggestion
The following example assumes you have a service listening on port 80. 
When accessed successfully, the service responds with `200 OK`. 
In the example, you'll configure the following:
* A service in {{site.base_gateway}} that points to the service you just deployed
* A route for this service
* Enable the JSON threat protection policy on the route to enforce payload limits. 
Specifically, you'll configure the plugin to reject violating requests.
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 18]
```diff
@@ -0,0 +1,238 @@
...
+
+Kong inspects the incoming requests, validates the payload adheres to the limits and rejects requests that violate the policy.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:42]:
```suggestion
Kong inspects the incoming requests, validates the payload adheres to the limits, and rejects requests that violate the policy.
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 19]
```diff
@@ -0,0 +1,238 @@
...
+  "response_buffering": true
+}
+```
+
+
+
+### Create a JSON-Threat-Protection Policy
+
+```bash
+curl -X POST http://127.0.0.1:8001/plugins \
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:129]:
```suggestion
curl -X POST http://localhost:8001/plugins \
```
We tend to use `localhost` across the docs for consistency.

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 20]
```diff
@@ -0,0 +1,238 @@
...
+- `config.max_container_depth=2`: The maximum depth of the container is 2.
+- `config.max_object_entry_count=4`: The number of object entries must not exceed 4.
+- `config.max_object_entry_name_length=7`: The key for an object entry must not exceed 7 bytes.
+- `config.max_array_element_count=2`: The number of array elements must not exceed 2.
+- `config.max_string_value_length=6`: The length of string values must not exceed 6 bytes.
+- `config.enforce_mode=block`: Enables `block` mode, where the request will not be proxied to the upstream service if the JSON violates the above limits.
+- `config.error_status_code=400`: When a JSON violation occurs, Kong returns an error code `400`.
+- `config.error_message=BadRequest1`: When a JSON violation occurs, Kong returns the error message `BadRequest1`.
+
+**Tap Mode**
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:155]:
```suggestion
### Tap mode
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 21]
```diff
@@ -0,0 +1,238 @@
...
+  "ordering": null,
+  "consumer_group": null,
+  "instance_name": null,
+  "enabled": true
+}
+```
+
+
+
+## Examples:
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:205]:
```suggestion
### Example requests and responses
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 22]
```diff
@@ -0,0 +1,238 @@
...
+- `config.max_object_entry_count=4`: The number of object entries must not exceed 4.
+- `config.max_object_entry_name_length=7`: The key for an object entry must not exceed 7 bytes.
+- `config.max_array_element_count=2`: The number of array elements must not exceed 2.
+- `config.max_string_value_length=6`: The length of string values must not exceed 6 bytes.
+- `config.enforce_mode=block`: Enables `block` mode, where the request will not be proxied to the upstream service if the JSON violates the above limits.
+- `config.error_status_code=400`: When a JSON violation occurs, Kong returns an error code `400`.
+- `config.error_message=BadRequest1`: When a JSON violation occurs, Kong returns the error message `BadRequest1`.
+
+**Tap Mode**
+In tap mode, the plugin will inspect the JSON in the request body, but if there are any violations of the limits, it will not block the request. Instead, it will log a warning and proxy the request to the upstream service. In other words, in tap mode, the plugin only monitors the traffic.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:156]:
```suggestion
In tap mode, the plugin inspects the JSON in the request body, but if there are any violations of the limits, it doesn't block the request. Instead, it logs a warning and proxies the request to the upstream service. In other words, in tap mode, the plugin only monitors the traffic.
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 23]
```diff
@@ -0,0 +1,238 @@
...
+
+If we want to enable Tap mode, we only need to set `config.enforce_mode` to `log_only`, like this:
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:158]:
```suggestion
To enable tap mode,  set `config.enforce_mode` to `log_only`:
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 24]
```diff
@@ -0,0 +1,238 @@
...
+
+```
+config.enforce_mode=log_only
+```
+
+
+After sending the above request, we will receive the following response:
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:165]:
```suggestion
The response will look something like this:
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 25]
```diff
@@ -0,0 +1,238 @@
...
+  "instance_name": null,
+  "enabled": true
+}
+```
+
+
+
+## Examples:
+
+This request is successfully proxied to returns a response back
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:207]:
```suggestion
This request is successfully proxied and returns a response:
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 26]
```diff
@@ -0,0 +1,238 @@
...
+
+```bash
+curl -XPOST http://127.1:8000/ \
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:210]:
```suggestion
curl -X POST http://localhost:8000/ \
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 27]
```diff
@@ -0,0 +1,238 @@
...
+
+The response is:
+
+```html
+200 OK
+```
+
+
+
+This request will be intercepted by the JSON-Threat-Protection plugin and will return the configured message.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:223]:
```suggestion

The following request is intercepted by the JSON Threat Protection plugin and returns the configured message instead of the standard response:
```

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 28]
```diff
@@ -0,0 +1,238 @@
...
+
+```bash
+curl -XPOST http://127.1:8000/ \
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:226]:
```suggestion
curl -X POST http://localhost:8000/ \
```

**Review by lena-larionova — CHANGES_REQUESTED** [review]

File: app/assets/images/icons/hub/kong-inc_json-threat-protection.png
Comment (lena-larionova) [inline app/assets/images/icons/hub/kong-inc_json-threat-protection.png:1]: Do you have the source file for this image, by any chance? Or is this file the source? There's some sizing/cropping issues that I can fix for you.

File: app/_src/.repos/kuma
Comment (lena-larionova) [inline app/_src/.repos/kuma:1]: This change shouldn't be in here, looks like a submodule update got accidentally pulled in?

File: app/_hub/kong-inc/json-threat-protection/_changelog.md [hunk 29]
```diff
@@ -0,0 +1,5 @@
+## Changelog
+
+### {{site.base_gateway}} 3.8.0.0
+
+* Introduced the new **JSON-Threat-Protection** plugin.
```
[the remark this answers is not in the document]
Comment (Water-Melon) [reply app/_hub/kong-inc/json-threat-protection/_changelog.md]: all resolved

**Review by veenaraja — APPROVED** [review]

File: app/_hub/kong-inc/json-threat-protection/overview/_index.md [hunk 30]
```diff
@@ -0,0 +1,258 @@
...
+
+
+Additionally, there is a limit named `max_body_size`, which is used to restrict the request body size.
+This can effectively reduce resource overhead and potential attack risks associated with excessively large bodies.
+The plugin reads the `Content-Length` header and compares its value with `max_body_size`.
+Therefore, in block mode, if the `Content-Length` header is missing or its value exceeds `max_body_size`, the request will be terminated.
+In tap mode, only the body size is checked, and logs are recorded.
+
+{:.note}
+> **Note**: max_body_size and nginx_http_client_max_body_size are independent of each other. Therefore, users should be aware that if nginx_http_client_max_body_size is set to a larger value while max_body_size is smaller and block mode is enabled, any request with a body size greater than max_body_size but less than nginx_http_client_max_body_size will be terminated.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/json-threat-protection/overview/_index.md:49]:
```suggestion
> **Note**: `max_body_size` and `nginx_http_client_max_body_size` are independent of each other. Therefore, if `nginx_http_client_max_body_size` is set to a larger value while `max_body_size` is smaller and block mode is enabled, any request with a body size greater than `max_body_size` but less than `nginx_http_client_max_body_size` will be terminated.
```
Sticking the params in backticks to appease the linter, and removing some extra wordiness.

**Review by lena-larionova — APPROVED** [review]
